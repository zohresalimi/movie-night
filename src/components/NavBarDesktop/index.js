import React, { useState } from "react";
import { Link } from "react-router-dom";
import { Icon, Menu, Container, Grid } from "semantic-ui-react";

export default function NavBarDesktop() {
  const [activeItem, setActiveItem] = useState("search");

  return (
    <Grid columns={1}>
      <Grid.Row>
        <Grid.Column>
          <Menu fixed="top" inverted>
            <Container>
              <Menu.Item
                as={Link}
                to="/"
                name="search"
                active={activeItem === "search"}
                onClick={() => setActiveItem("search")}
              >
                <Icon name="search" />
                Search
              </Menu.Item>
              <Menu.Item
                as={Link}
                to="/watch-later"
                name="watch later"
                active={activeItem === "watch later"}
                onClick={() => setActiveItem("watch later")}
              >
                <Icon name="clock" />
                Watch Later
              </Menu.Item>
              <Menu.Item
                as={Link}
                to="/favorites"
                name="favorites"
                data-testid="favorites"
                active={activeItem === "favorites"}
                onClick={() => setActiveItem("favorites")}
              >
                <Icon name="star" />
                Favorites
              </Menu.Item>
            </Container>
          </Menu>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  );
}
