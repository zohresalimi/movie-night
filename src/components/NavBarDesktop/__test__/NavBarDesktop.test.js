import * as React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render, fireEvent, act } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import NavBarDesktop from "../";

async function renderWrapper() {
  let component;

  act(() => {
    component = render(
      <Router>
        <NavBarDesktop />
      </Router>
    );
  });

  return component;
}

describe("NavBarDesktop Component Testing", () => {
  test("take snapshot", async () => {
    const { container } = await renderWrapper();
    expect(container).toMatchSnapshot();
  });

  test("match snapshot if active item is changed", async () => {
    const { container, getByTestId } = await renderWrapper();
    const menuItem = getByTestId("favorites");
    fireEvent.click(menuItem);
    expect(container).toMatchSnapshot();
  });
});
