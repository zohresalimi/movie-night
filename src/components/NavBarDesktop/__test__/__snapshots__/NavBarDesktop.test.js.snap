// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarDesktop Component Testing match snapshot if active item is changed 1`] = `
<div>
  <div
    class="ui one column grid"
  >
    <div
      class="row"
    >
      <div
        class="column"
      >
        <div
          class="ui inverted top fixed menu"
        >
          <div
            class="ui container"
          >
            <a
              class="item"
              href="/"
            >
              <i
                aria-hidden="true"
                class="search icon"
              />
              Search
            </a>
            <a
              class="item"
              href="/watch-later"
            >
              <i
                aria-hidden="true"
                class="clock icon"
              />
              Watch Later
            </a>
            <a
              class="active item"
              data-testid="favorites"
              href="/favorites"
            >
              <i
                aria-hidden="true"
                class="star icon"
              />
              Favorites
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NavBarDesktop Component Testing take snapshot 1`] = `
<div>
  <div
    class="ui one column grid"
  >
    <div
      class="row"
    >
      <div
        class="column"
      >
        <div
          class="ui inverted top fixed menu"
        >
          <div
            class="ui container"
          >
            <a
              class="active item"
              href="/"
            >
              <i
                aria-hidden="true"
                class="search icon"
              />
              Search
            </a>
            <a
              class="item"
              href="/watch-later"
            >
              <i
                aria-hidden="true"
                class="clock icon"
              />
              Watch Later
            </a>
            <a
              class="item"
              data-testid="favorites"
              href="/favorites"
            >
              <i
                aria-hidden="true"
                class="star icon"
              />
              Favorites
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
